### PR TITLE
New token claim for a specific organization attribute

### PR DESCRIPTION
**Pull request by prifulnath**
While doing some organization attribute mapping using "Organization Attribute" mapper. Some of the confidential organization attribute information was also added in the token. So, I added an organization specific single attribute selection mapper in this PR.

New mapper is "Organization Specific Attribute", which will add the attribute matching claim name, just as the user attribute.

Discussion about my [issue](https://github.com/p2-inc/phasetwo-containers/issues/42#issuecomment-2084799573), You can find little more info here.

### CODE REVIEW

**Comment by xgp** [comment]
@prifulnath Thanks for the PR. This will need a test before we can merge. See https://github.com/p2-inc/keycloak-orgs/blob/main/src/test/java/io/phasetwo/service/mapper/OrganizationRoleMapperTest.java for an example.

**Comment by prifulnath** [comment]
@xgp Added test for the new claim mapper.

**Comment by xgp** [comment]
@rtufisi Can also review this PR? Specifically see if the tests are thorough and test for edge cases (e.g. multiple attributes, multi-valued attributes, null value attributes, etc.)?

File: src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          configProperties);
+    }
+  
+    @Override
+    protected Map<String, Object> getOrganizationClaim(
+        KeycloakSession session, RealmModel realm, UserModel user, ProtocolMapperModel mappingModel) {
+      OrganizationProvider orgs = session.getProvider(OrganizationProvider.class);
+      Map<String, Object> claim = Maps.newHashMap();
+      orgs.getUserOrganizationsStream(realm, user)
+          .forEach(
```
Comment (rtufisi) [inline src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java:46]: Maybe we can filter if a attribute is missing in a organization?

File: src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          TOKEN_MAPPER_CATEGORY,
+          "Map organization single specific attributes in a token claim.",
+          configProperties);
+    }
+  
+    @Override
+    protected Map<String, Object> getOrganizationClaim(
+        KeycloakSession session, RealmModel realm, UserModel user, ProtocolMapperModel mappingModel) {
+      OrganizationProvider orgs = session.getProvider(OrganizationProvider.class);
+      Map<String, Object> claim = Maps.newHashMap();
```
Comment (rtufisi) [inline src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java:44]: we should name this variable organizationClaim. To be more specific what is the key-value relationship

File: src/test/java/io/phasetwo/service/mapper/OrganizationSpecificAttributeMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+@JBossLog
+class OrganizationSpecificAttributeMapperTest extends AbstractOrganizationTest {
+
+  public static final String CLAIM = "secret_attr";
+
+  @Test
+  void shouldConfigureOrganizationSpecificAttributeMapperOidcProtocolMapper() throws Exception {
+    // add organization with attribute 'secret_attr'
+    var organization = createOrganization(
```
Comment (rtufisi) [inline src/test/java/io/phasetwo/service/mapper/OrganizationSpecificAttributeMapperTest.java:38]: Could we add a second organization to the set of user organizations? In the second organization I would not add the CLAIM to test what is the behaviour if the claim is not available.

**Review by rtufisi — COMMENTED** [review]
Great job with creating this new mapper

File: src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java (hunk 1 above)
Comment (prifulnath) [reply src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java]: Added filter for null value.

`
orgs.getUserOrganizationsStream(realm, user)
        .forEach(
            o -> {
              // add to token only when value is available
              String attributeValue = o.getFirstAttribute(mappingModel.getName());
              if (attributeValue != null) {
                organizationClaim.put(o.getId(), attributeValue);
              }
            });
`

File: src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java (hunk 2 above)
Comment (prifulnath) [reply src/main/java/io/phasetwo/service/protocol/oidc/mappers/OrganizationSpecificAttributeMapper.java]: changed the name to `organizationClaim`

File: src/test/java/io/phasetwo/service/mapper/OrganizationSpecificAttributeMapperTest.java (hunk 3 above)
Comment (prifulnath) [reply src/test/java/io/phasetwo/service/mapper/OrganizationSpecificAttributeMapperTest.java]: Added 4 organization and added one extra claim mapper for the validation.
- Organization 1 with secret_attr value My Secret Value
- Organization 2 with secret_attr value My Second Secret Value
- Organization 3 with secret_attr value empty string
- Organization 4 without secret_attr value and another attribute another_attribute with value  My Value

**Review by xgp — APPROVED** [review]
